apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -545,23 +579,27 @@ impl std::fmt::Display for CachingQueryKey {
         hasher.update(
             &serde_json::to_vec(&self.plan_options).expect("serialization should not fail"),
         );
+        hasher
+            .update(&serde_json::to_vec(&self.config_mode).expect("serialization should not fail"));
         hasher.update(&serde_json::to_vec(&self.sdl).expect("serialization should not fail"));
         let metadata = hex::encode(hasher.finalize());
 
         write!(
             f,
-            "plan:{}:{}:{}:{}",
-            FEDERATION_VERSION, self.hash, operation, metadata,
+            "plan:{}:{}:{}:{}:{}",
+            CACHE_KEY_VERSION, FEDERATION_VERSION, self.hash, operation, metadata,
```

Comment:
It's unrelated to this particular PR, but it took me a while to understand that `operation` is a hash of `self.operation`, which is actually an operation name.

Also, `self.hash` is hash of query and relevent parts of schema, but taking into account only operation matching `operation_name`.
At the same time, `self.hash` doesn't include a hash of "operation_name" itself

Reply:
Also `self.metadata` is actually "authorization metadata".

P.S. Same as the previous comment, this is unrelated to this PR.
Just comments for others to also understand what is happening in this code.

Reply:
I'd like to see this refactored and keys become self describing.